graphqlapi/src/main/java/io/stargate/graphql/schema/DmlSchemaBuilder.java

```diff
@@ -63,61 +57,29 @@
 
   private final Persistence persistence;
   private final AuthenticationService authenticationService;
-  private final Map<Column.ColumnType, GraphQLInputObjectType> filterInputTypes;
   private final Map<Table, GraphQLOutputType> entityResultMap = new HashMap<>();
-  private final Map<String, GraphQLType> typeCache = new HashMap<>();
+  private final FieldInputTypeCache fieldInputTypes;
+  private final FieldOutputTypeCache fieldOutputTypes;
+  private final FieldFilterInputTypeCache fieldFilterInputTypes;
   private final NameMapping nameMapping;
   private final Keyspace keyspace;
 
-  private GraphQLInputType mutationOptions =
-      GraphQLInputObjectType.newInputObject()
-          .name("MutationOptions")
-          .field(
-              GraphQLInputObjectField.newInputObjectField()
-                  .name("consistency")
-                  .type(
-                      GraphQLEnumType.newEnum()
-                          .name("MutationConsistency")
-                          .value("LOCAL_ONE")
-                          .value("LOCAL_QUORUM")
-                          .value("ALL")
-                          .build())
-                  .build())
-          .field(
-              GraphQLInputObjectField.newInputObjectField()
-                  .name("serialConsistency")
-                  .type(
-                      GraphQLEnumType.newEnum()
-                          .name("SerialConsistency")
-                          .value("SERIAL")
-                          .value("LOCAL_SERIAL")
-                          .build())
-                  .build())
-          .field(
-              GraphQLInputObjectField.newInputObjectField()
-                  .name("ttl")
-                  .type(Scalars.GraphQLInt)
-                  .defaultValue(-1)
-                  .build())
-          .build();
-
   DmlSchemaBuilder(
       Persistence persistence, AuthenticationService authenticationService, Keyspace keyspace) {
     this.persistence = persistence;
     this.authenticationService = authenticationService;
     this.keyspace = keyspace;
 
     this.nameMapping = new NameMapping(keyspace.tables(), keyspace.userDefinedTypes());
-    this.filterInputTypes = buildFilterInputTypes();
+    this.fieldInputTypes = new FieldInputTypeCache(this.nameMapping);
+    this.fieldOutputTypes = new FieldOutputTypeCache(this.nameMapping);
+    this.fieldFilterInputTypes =
+        new FieldFilterInputTypeCache(this.fieldInputTypes, this.nameMapping);
```

Comment:
I think we have to consider that names should be "granted" or generated by the `NameMapping` instance.

Given that UDTs, map entries, tables GraphQL names can collide, a central entity generating those names in a deterministic way (i.e. UDTs first, then maps and last tables), with suffixes and prefixes included.

We did something like that on the golang project: see [here](https://github.com/datastax/cassandra-data-apis/blob/0f24c6a5f6ca28c0086c49ad0ce2997994e7f841/config/naming.go#L72), [here](https://github.com/datastax/cassandra-data-apis/blob/0f24c6a5f6ca28c0086c49ad0ce2997994e7f841/config/naming.go#L63) and [this unit test](https://github.com/datastax/cassandra-data-apis/blob/0f24c6a5f6ca28c0086c49ad0ce2997994e7f841/config/naming_test.go#L50-L61) for example .

I see a strong relationship between naming and the field cache you propose in this PR, so maybe it makes sense to address them in one big bang, i.e. go over DB objects, generate names and graphql (input/output) types for them.

Reply:
> I think we have to consider that names should be "granted" or generated by the NameMapping instance

They already are, at least the parts that are relevant to the CQL->GraphQL conversion. The fact that we have "Input" and "FilterInput" types, and generate "EntryXKeyYValue" types for maps, are purely specific to GraphQL, so I'm less concerned having these rules directly in the GraphQL code. But if you prefer to extract them into utility methods, I'm not opposed either.

> UDTs, map entries, tables GraphQL names can collide

There are different levels of collision:
1. two tables `foo_bar` and `"FooBar"` can collide, that's the discussion I started in #216
2. a table `foo_bar` and a UDT `foo_bar` do not collide, thanks to the `Udt` suffix.
3. I think the current naming rules do not allow different CQL column types to collide, see [this above comment](https://github.com/stargate/stargate/pull/286#discussion_r506771839).
4. then there are the corner cases where someone might name a table `foo_bar_udt` or `entry_int_key_string_value`, I think we can skip those and log a warning somewhere.

> We did something like that on the golang project

Note that the strategy used in that  project is not resilient to schema changes, one thing to keep in mind if we adapt it.

Reply:
I've posted a new comment on the naming discussion https://github.com/stargate/stargate/pull/216#issuecomment-713408183 .

This is not directly related to this change, IMO a refactoring like this would have been an opportunity to also tackle cross cutting concept like graphql naming. Otherwise, it's very likely than a new refactor is needed to deliver that feature.